cromwell-drs-localizer/src/main/scala/drs/localizer/downloaders/BulkAccessUrlDownloader.scala

```diff
@@ -40,39 +43,25 @@ case class BulkAccessUrlDownloader(resolvedUrls : List[ResolvedDrsUrl]) extends
     * @return Filepath of a getm-manifest.json that Getm can use to download multiple files in parallel.
     */
   def generateJsonManifest(resolvedUrls : List[ResolvedDrsUrl]): IO[Path] = {
-    def toJsonString(drsResponse: DrsResolverResponse, destinationFilepath: String): String = {
-      //NB: trailing comma is being removed in generateJsonManifest
-      val accessUrl: AccessUrl = drsResponse.accessUrl.getOrElse(AccessUrl("missing", None))
-      drsResponse.hashes.map(_ => {
-        val checksum = GetmChecksum(drsResponse.hashes, accessUrl).value.getOrElse("error_calculating_checksum")
-        val checksumAlgorithm = GetmChecksum(drsResponse.hashes, accessUrl).getmAlgorithm
-        s"""  {
-           |    "url" : "${accessUrl.url}",
-           |    "filepath" : "$destinationFilepath",
-           |    "checksum" : "$checksum",
-           |    "checksum-algorithm" : "$checksumAlgorithm"
-           |  },
-           |""".stripMargin
-      }).getOrElse(
-        s"""  {
-           |    "url" : "${accessUrl.url}",
-           |    "filepath" : "$destinationFilepath"
-           |  },
-           |""".stripMargin
-      )
-    }
-    IO {
-      var jsonString: String = "[\n"
-      for (resolvedUrl <- resolvedUrls) {
-        jsonString += toJsonString(resolvedUrl.drsResponse, resolvedUrl.downloadDestinationPath)
-      }
-      if(jsonString.contains(',')) {
-        //remove trailing comma from array elements, but don't crash on empty list.
-        jsonString = jsonString.substring(0, jsonString.lastIndexOf(","))
-      }
-      jsonString += "\n]"
-      Files.write(getmManifestPath, jsonString.getBytes(StandardCharsets.UTF_8))
+    def resolvedUrlToJsonMap(resolvedUrl: ResolvedDrsUrl): Map[String,String] = {
+      val accessUrl: AccessUrl = resolvedUrl.drsResponse.accessUrl.getOrElse(AccessUrl("missing", None))
+      resolvedUrl.drsResponse.hashes.map{_ =>
+        val checksum = GetmChecksum(resolvedUrl.drsResponse.hashes, accessUrl).value.getOrElse("error_calculating_checksum")
+        val checksumAlgorithm = GetmChecksum(resolvedUrl.drsResponse.hashes, accessUrl).getmAlgorithm
+        Map(
+          ("url", accessUrl.url),
+          ("filepath", resolvedUrl.downloadDestinationPath),
+          ("checksum", checksum),
+          ("checksum-algorithm", checksumAlgorithm)
+        )
+      }.getOrElse(Map(
+        ("url", accessUrl.url),
+        ("filepath", resolvedUrl.downloadDestinationPath)
+      ))
     }
+
+    val jsonArray: String = resolvedUrls.map(resolved => resolvedUrlToJsonMap(resolved)).toJson.prettyPrint
```

Comment:
Interesting question! Would that bring us back to manual json-string-munging so we could stream bytes into a file?? ;) 

Back of the napkin math: a *very* long DRS URL might be 1,000 characters. Java strings are 40 + length bytes. Call it 1kb per DRS url. 

100MB of RAM gets us ~100,000 DRS URLs. Seems like we would need some pretty extreme inputs to cause a problem.

Reply:
I think it would probably bring us to a streaming write using Spray - surely there's a way to stream bytes into an open files using `IO`. Seems like not something we'll need to worry about in the near future, though!